invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -111,9 +111,12 @@ def _log_errors(errors):
         # DataCiteError is a tuple with the errors on the first
         errors = json.loads(errors.args[0])["errors"]
         for error in errors:
-            field = error["source"]
+            field = error.get("source", None)
```

Comment:
"source" is not always present

Reply:
I'm wondering if it ever is, or if we should be using another field. At this point we are logging https://github.com/inveniosoftware/datacite/blob/master/datacite/errors.py which do not seem to enforce any. Maybe @tmorrell can guide us here.

As temporary measure I think doing the `.get()` is okay, but we need to understand what is the alternative field and if it is correct.

Reply:
"source" will show up if there is an issue with a specific metadata field, but not if there is a general error like incorrect credentials or a missing prefix. So I think the get is right, but the 'Missing field' text below is a bit confusing because you can get errors for reasons other than missing fields. 

One possible other improvement is the 403 error contains nothing useful in the `text`field beyond You are not authorized to access this resource. Maybe returning something like "403 Error: Check your credentials and prefix settings" would he more helpful for folks.